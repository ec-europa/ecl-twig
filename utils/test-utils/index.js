const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
