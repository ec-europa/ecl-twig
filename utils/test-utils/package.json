{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
