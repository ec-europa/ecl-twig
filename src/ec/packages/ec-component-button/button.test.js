import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import dataPrimary from '@ecl/ec-specs-button/demo/data--primary';
import dataSecondary from '@ecl/ec-specs-button/demo/data--secondary';
import dataCall from '@ecl/ec-specs-button/demo/data--call';
import dataGhost from '@ecl/ec-specs-button/demo/data--ghost';
import dataSearch from '@ecl/ec-specs-button/demo/data--search';

describe('EC - Button', () => {
  const template = path.resolve(__dirname, './button.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  describe('Primary', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataPrimary)).resolves.toMatchSnapshot();
    });
  });

  describe('Secondary', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataSecondary)).resolves.toMatchSnapshot();
    });
  });

  describe('CTA', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      dataCall.icon.path = 'static/icons.svg';
      dataCall.icon.type = 'ui';
      dataCall.icon.name = 'corner-arrow';
      dataCall.icon.size = 'fluid';

      return expect(render(dataCall)).resolves.toMatchSnapshot();
    });
  });

  describe('Ghost', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataGhost)).resolves.toMatchSnapshot();
    });
  });

  describe('Search', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataSearch)).resolves.toMatchSnapshot();
    });
  });

  describe('CTA button - icon before', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      dataCall.icon.path = 'static/icons.svg';
      dataCall.icon.type = 'ui';
      dataCall.icon.name = 'corner-arrow';
      dataCall.icon.size = 'fluid';
      dataCall.icon_position = 'before';

      return expect(render(dataCall)).resolves.toMatchSnapshot();
    });
  });

  describe('CTA button - icon after', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      const buttonData = {
        label: 'CTA Button with icon',
        variant: 'call',
        icon: {
          path: 'static/icons.svg',
          type: 'ui',
          name: 'corner-arrow',
          size: 'fluid',
        },
      };

      return expect(render(buttonData)).resolves.toMatchSnapshot();
    });
  });
});
