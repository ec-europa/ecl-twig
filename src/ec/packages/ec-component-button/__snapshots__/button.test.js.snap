// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Button CTA button - icon after renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--call\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <span class=\\"ecl-button__label\\">CTA Button with icon</span>

    <svg
      class=\\"ecl-icon ecl-icon--fluid ecl-button__icon ecl-button__icon--after\\"
      focusable=\\"false\\"
      aria-hidden=\\"true\\"
    >
      <use xlink:href=\\"static/icons.svg#ui--corner-arrow\\"></use>
    </svg>
  </span>
</button>
"
`;

exports[`EC - Button CTA button - icon before renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--call\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <svg
      class=\\"ecl-icon ecl-icon--fluid ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--before\\"
      focusable=\\"false\\"
      aria-hidden=\\"true\\"
    >
      <use xlink:href=\\"static/icons.svg#ui--corner-arrow\\"></use>
    </svg>
    <span class=\\"ecl-button__label\\">Call to action button</span>
  </span>
</button>
"
`;

exports[`EC - Button CTA renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--call\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <span class=\\"ecl-button__label\\">Call to action button</span>

    <svg
      class=\\"ecl-icon ecl-icon--fluid ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after\\"
      focusable=\\"false\\"
      aria-hidden=\\"true\\"
    >
      <use xlink:href=\\"static/icons.svg#ui--corner-arrow\\"></use>
    </svg>
  </span>
</button>
"
`;

exports[`EC - Button Ghost renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--ghost\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <span class=\\"ecl-button__label\\">Ghost button</span>
  </span>
</button>
"
`;

exports[`EC - Button Primary renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--primary\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <span class=\\"ecl-button__label\\">Primary button</span>
  </span>
</button>
"
`;

exports[`EC - Button Search renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--search\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <span class=\\"ecl-button__label\\">Search button</span>
  </span>
</button>
"
`;

exports[`EC - Button Secondary renders correctly 1`] = `
"<button class=\\"ecl-button ecl-button--secondary\\" type=\\"submit\\">
  <span class=\\"ecl-button__container\\">
    <span class=\\"ecl-button__label\\">Secondary button</span>
  </span>
</button>
"
`;
